mathesar_ui/src/stores/abstract-types/operations/filtering.ts

```diff
@@ -218,13 +219,14 @@ export const filterDefinitionMap = getFilterDefinitionMap();
 
 export function getFiltersForAbstractType(
   categoryIdentifier: AbstractTypeCategoryIdentifier,
+  isFK: FkConstraint
 ): Map<AbstractTypeFilterDefinition['id'], AbstractTypeFilterDefinition> {
   const allowedFiltersMap: Map<
     AbstractTypeFilterDefinition['id'],
     AbstractTypeFilterDefinition
   > = new Map();
 
-  filterDefinitionMap.get(categoryIdentifier)?.forEach((filter) => {
+  filterDefinitionMap.get((isFK !== undefined ? "boolean"  : categoryIdentifier))?.forEach((filter) => {
     allowedFiltersMap.set(filter.id, filter);
   });
```

Comment:
@LFLATH 
As mentioned on [the comment in the issue](https://github.com/centerofci/mathesar/issues/1523#issuecomment-1322381116), these change should not be in `filtering.ts`.

It should be in a utility file as part of the filter-entry component, similar to the cell-fabric component.